corehq/apps/hqcase/api/updates.py

```diff
@@ -24,13 +24,25 @@ class JsonIndex(jsonobject.JsonObject):
     case_id = jsonobject.StringProperty()
     external_id = jsonobject.StringProperty()
     temporary_id = jsonobject.StringProperty()
-    case_type = jsonobject.StringProperty(required=True)
-    relationship = jsonobject.StringProperty(required=True, choices=('child', 'extension'))
+    case_type = jsonobject.StringProperty()
+    relationship = jsonobject.StringProperty(choices=('child', 'extension'))
 
     def validate(self, *args, **kwargs):
-        super().validate(*args, **kwargs)
-        if len(list(filter(None, [self.case_id, self.external_id, self.temporary_id]))) != 1:
+        ids_specified = len(list(filter(None, [self.case_id, self.external_id, self.temporary_id])))
+        if ids_specified > 1:
             raise BadValueError("Indices must specify case_id, external_id, or temporary ID, and only one")
+        if ids_specified == 1:
+            self.properties()['case_type'].required = True
+            self.properties()['relationship'].required = True
+        super().validate(*args, **kwargs)
+
+    def get_id(self, case_db):
+        if self.temporary_id:
+            return case_db.lookup_id(self.temporary_id, 'temporary_id')
+        if self.external_id:
+            return case_db.lookup_id(self.external_id, 'external_id')
+        # case_id may be unspecified, which is fine - that's how deletions work
+        return self.case_id
```

Comment:
One thing I don't love about this approach is we can't distinguish between `{}` and `{'case_id': ""}`.  I think that's okay though.

Reply:
You could check in the `wrap` method and force the data to contain `case_id`. I agree it's not a blocker